src/chainparamsbase_settings.h

```diff
@@ -0,0 +1,56 @@
+#ifndef BITCOIN_CHAINPARAMSBASE_SETTINGS_H
+#define BITCOIN_CHAINPARAMSBASE_SETTINGS_H
+
+#include <chainparamsbase.h>
+#include <common/setting.h>
+
+#include <string>
+#include <vector>
+
+using SignetseednodeSetting = common::Setting<
```

Comment:
The part that bothers me is that this seems "Stringly-Typed" this way, global scoped and the user doesn't even have proper code completion since there isn't a common prefix or namespace or enum to narrow the options for code completion as far as I can tell.
I'm not recommending a top level settings (but even that would be better than these globals), but rather targeted nested namespaces (e.g., wallet::setting, init::setting) or something that would group these in a way that the compiler understands instead of just having a naming convention.

> Also BlocksDirSetting::Get reads better to me than Setting::BlocksDir::Get

What about inverting it and making the `Get` method accept different types based on the context, something like: `Init::Setting::Get<TestActivationHeight>(args)`? I haven't thought this through (I can't meaningfully review a change this big, we need to find a way to narrow it down), I'm just wondering how we could make the compiler work for us instead of coming up with special names.